### PR TITLE
feat: add axios mocking to component test-suite

### PR DESCRIPTION
**Pull request by sanpj2292**
## Description of the change

Resolves INT-371

Axios mocking capability in component test-suite

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: test/integrations/component.test.ts [hunk 1]
```diff
@@ -34,6 +38,68 @@ afterAll(async () => {
...
+      return retVal(key);
+    };
+  }
+  
+  // post, patch, put
+  return (url, data, opts) => {
+    // mock result from some cache
+    const key = stringify({ url, data, ...opts })
+    return retVal(key);
+  };
```
Comment (ItsSudip) [inline test/integrations/component.test.ts:87]: Can't we have one common function apart from three? Which will stringify url, data and opts always.

**Review by ItsSudip — DISMISSED** [review]

File: test/integrations/component.test.ts [hunk 2]
```diff
@@ -34,6 +38,68 @@ afterAll(async () => {
   await createHttpTerminator({ server }).terminate();
 });
 
+// unmock already existing axios-mocking
+jest.unmock('axios')
+
+jest.mock('axios')
```
Comment (koladilip) [inline test/integrations/component.test.ts:44]: https://www.npmjs.com/package/axios-mock-adapter this looks interesting, should we try this?

File: test/integrations/component.test.ts [hunk 3]
```diff
@@ -111,6 +177,13 @@ const sourceTestHandler = async (tcData) => {
   )}`;
   await testRoute(route, tcData);
 };
+// all the axios requests will be stored in this map
+const allAxiosReqsMap = allTestDataFilePaths.reduce((agg, currPath) => {
```
Comment (koladilip) [inline test/integrations/component.test.ts:181]: should we build this per test file and reset again? example: when we running pardot then it it has only maps build for it and reset after it is completed, is it easy to do?

**Review by koladilip — APPROVED** [review]

File: test/integrations/component.test.ts (hunk 3 above)
Comment (sanpj2292) [reply test/integrations/component.test.ts]: Don't think it is easily possible

File: test/integrations/component.test.ts (hunk 1 above)
Comment (sanpj2292) [reply test/integrations/component.test.ts]: Will try to see if something of that sort is possible.

**Comment by sanpj2292** [comment]
Resolves INT-371

File: test/integrations/component.test.ts (hunk 3 above)
Comment (koladilip) [reply test/integrations/component.test.ts]: then it is fine.

File: test/integrations/component.test.ts (hunk 2 above)
Comment (sanpj2292) [reply test/integrations/component.test.ts]: Used this and implemented the network requests mocking

Comment (sanpj2292) [reply test/integrations/component.test.ts]: Hence resolving this comment

File: test/integrations/component.test.ts (hunk 1 above)
Comment (sanpj2292) [reply test/integrations/component.test.ts]: The approach has been changed. Hence resolving this comment

**Review by koladilip — DISMISSED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**